### PR TITLE
feat(protocol): Change require to custom err in bridge contracts

### PR DESCRIPTION
**Pull request by adaki2004**
As per issue:
https://github.com/taikoxyz/taiko-mono/issues/13057

Bridge related require messages are changed in code + unit tests.

Created a gas usage comparison between current version vs. new. Please see differences in yellow (cheaper) or purple (expensive). Left is the new, right is the old.
Képernyőfotó 2023-02-27 - 9 35 07

It might only be slightly more expensive because only called 4 times.

### CODE REVIEW

File: packages/protocol/contracts/bridge/BridgeCustomErrors.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.18;
+
+abstract contract BridgeCustomErrors {
+    // Libraries having custom errors in place
+    error B_CANNOT_RECEIVE();
```
Comment (dantaik) [inline packages/protocol/contracts/bridge/BridgeCustomErrors.sol:11]: sort these errors?

File: packages/protocol/contracts/bridge/BridgeCustomErrors.sol [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.18;
+
+abstract contract BridgeCustomErrors {
+    // Libraries having custom errors in place
```
Comment (davidtaikocha) [inline packages/protocol/contracts/bridge/BridgeCustomErrors.sol:10]: I believe the reason why we want to define a `BridgeCustomErrors` is that we want all custom errors which may be encountered when calling `Bridge` contract to be a part of the `Bridge` contract's compiled ABI, which will be helpful for the clients that only import the `Bridge` ABI to parse those errors. so i think we should also add those custom errors defined in `LibBridgeXXX` to here?

Comment (adaki2004) [reply packages/protocol/contracts/bridge/BridgeCustomErrors.sol]: Make sense and done.

**Review by davidtaikocha — DISMISSED** [review]

File: packages/protocol/contracts/bridge/BridgeCustomErrors.sol (hunk 1 above)
Comment (adaki2004) [reply packages/protocol/contracts/bridge/BridgeCustomErrors.sol:10]: Done.

**Comment by cyberhorsey** [comment]
LGTM just needs merge conflict resolved!

**Review by davidtaikocha — DISMISSED** [review]

**Comment by adaki2004** [comment]
Integration tests has to be adjusted.

**Review by cyberhorsey — DISMISSED** [review]

**Review by davidtaikocha — DISMISSED** [review]

**Review by dantaik — APPROVED** [review]

**Review by cyberhorsey — APPROVED** [review]

**Pull request merged**